src/compiler/checker.ts

```diff
@@ -31385,7 +31385,8 @@ namespace ts {
                     reportImplicitAny(declaration, anyType);
                     return anyType;
                 }
-                else if (isEmptyArrayLiteralType(widened)) {
+
+                if (isEmptyArrayLiteralType(widened)) {

```

Comment:
just checking, but...this diff doesn't change anything, right?

Reply:
Later: I tested locally that it doesn't.

Reply:
@sandersn Yes, it doesn't change anything, it seems I just forgot to return the `else` after debugging. I reverted that change back. Sorry for the confusion.